### ISSUE TITLE
Values are different in Pytorch and C++

### ISSUE THREAD

**Issue by Sunny-DV** [alantess/learntorch#1]
Hi, Thanks for the amazing tutorials. 
I am solving 1D pde and tried to call that  model in C++. I have done this successfully but the values i am predicting in Pytorch and in C++ are different. 

#include <torch/script.h>
#include <memory>
#include <iostream>
int main(int argc, const char* argv[])
{
 if (argc != 2) {
    std::cerr << "usage: main <path-to-exported-script-module>\n";
    return -1;
  }

  try {
    // Deserialize the ScriptModule from a file using torch::jit::load().
    torch::jit::script::Module module;
    module = torch::jit::load("/home/sunny/neural/diffusion.pth");
    // Create a vector of inputs.
    double x;
    std::cout<<"Please enter the value of x ";
    std::cin>>x;
    double t;
    std::cout<<"Please enter the value of t ";
    std::cin>>t;
    
    std::vector<torch::jit::IValue> inputs;
    inputs.push_back(torch::tensor({{x,t}}));
    // Execute the model and turn its output into a tensor.
    torch::Tensor output = module.forward((inputs)).toTensor();

    std::cout << output << std::endl;
    
  }
  catch (const c10::Error& e) {
    std::cerr << "error loading the model\n";
    return -1;
  }

}

I used the same method you mentioned to save the model. Can you tell me what i am doing wrong?

**Comment by alantess** [comment]
Did you normalize the values before running the model? A Tensor is automatically sets values between 0 to 1.

**Comment by Sunny-DV** [comment]
Actually when i was saving the trained model i have not freeze the weights that's the reason i got different results.

**Comment by alantess** [comment]
Closing issue
